crates/database/src/orders.rs

```diff
@@ -277,6 +253,69 @@ pub fn is_duplicate_record_error(err: &sqlx::Error) -> bool {
     false
 }
 
+/// Time during the `settle()` call when an interaction should be executed.
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, sqlx::Type)]
+#[sqlx(type_name = "ExecutionTime")]
+#[sqlx(rename_all = "lowercase")]
+pub enum ExecutionTime {
+    /// Interaction that gets executed before sell tokens get transferred from
+    /// user to settlement contract.
+    #[default]
+    Pre,
+    /// Interaction that gets executed after buy tokens got sent from the
+    /// settlement contract to the user.
+    Post,
+}
+
+/// One row in the `interactions` table.
+#[derive(Clone, Debug, Default, Eq, PartialEq, sqlx::FromRow)]
+pub struct Interaction {
+    pub target: Address,
+    pub value: BigDecimal,
+    pub data: Vec<u8>,
+    pub index: i32,
+    pub execution: ExecutionTime,
+}
+
+pub async fn insert_interactions(
+    ex: &mut PgConnection,
+    order: &OrderUid,
+    interactions: &[Interaction],
+) -> Result<(), sqlx::Error> {
+    for interaction in interactions {
+        insert_interaction(ex, order, interaction).await?;
+    }
+    Ok(())
+}
+
+pub async fn insert_interaction(
+    ex: &mut PgConnection,
```

Comment:
This file also has `fn insert_or_overwrite_interaction`. It feels like they should be next to eachother.

Reply:
614890ad82304118ca91bb21742d95c540fbded6